### PR TITLE
[5183] Warning message about saving invalid XML files is missing in version 4.

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/5174
https://github.com/craftercms/craftercms/issues/5183

### CODE REVIEW

File: ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx [hunk 1]
```diff
@@ -184,7 +185,21 @@ export default function SiteConfigurationManagement(props: SiteConfigurationMana
...
+          message: formatMessage(translations.xmlContainsErrors, {
+            errors: doc
+          }),
+          options: {
+            variant: 'error'
+          }
+        })
+      );
+      return;
+    }
```
Comment (rart) [inline ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx]: extract to local function to use in both places

File: ui/app/src/index.tsx [hunk 2]
```diff
@@ -28,3 +28,4 @@ if (process.env.NODE_ENV === 'production') {
   const elem = document.getElementById('root');
   ReactDOM.render(<App />, elem);
 }
+export { parseValidateDocument } from './utils/xml';
```
Comment (rart) [inline ui/app/src/index.tsx]: why? rollback

File: ui/app/src/components/PluginConfigDialog/PluginConfigDialogContainer.tsx [hunk 3]
```diff
@@ -93,7 +108,7 @@ export function PluginConfigDialogContainer(props: PluginConfigDialogContainerPr
           onSaved();
         },
         error: ({ response }) => {
-          onSubmittingAndOrPendingChange({ isSubmitting: false });
+          onSubmittingAndOrPendingChange({ isSubmitting: false, hasPendingChanges: true });
```
Comment (rart) [inline ui/app/src/components/PluginConfigDialog/PluginConfigDialogContainer.tsx]: It feels like by this point it should already be true. Why do you need to set it here?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/hooks/useEnhancedDialogState.ts [hunk 4]
```diff
@@ -75,5 +85,5 @@ export function useEnhancedDialogState(initialState?: Partial<EnhancedDialogStat
       onHasPendingChange,
       onSubmittingAndOrPendingChange
     };
-  }, [setState, state]);
+  }, [initialState, setState, state]);
```
Comment (rart) [inline ui/app/src/hooks/useEnhancedDialogState.ts]: This is dangerous to include here as users of the effect most likely won’t memo the initial state. Ignore it and add a comment to justify the eslint ignore

File: ui/app/src/utils/xml.ts [hunk 5]
```diff
@@ -181,3 +181,14 @@ export function deserialize(xml: string | Node, options?: X2jOptionsOptional): a
 export function cdataWrap(value: string): string {
   return `<![CDATA[${value}]]>`;
 }
+
+export function parseValidateDocument(content: string): XMLDocument | string {
```
Comment (rart) [inline ui/app/src/utils/xml.ts:185]: Is the parsed document ever used for something? I’m thinking the function should have a single responsibility.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/utils/xml.ts (hunk 5 above)
Comment (aachavarria) [reply ui/app/src/utils/xml.ts:185]: yes, the doc is used

**Review by rart — APPROVED** [review]

**Pull request merged**